### ISSUE TITLE
Loss of the generator

### ISSUE THREAD

**Issue by rylanchiu** [LantaoYu/SeqGAN#10]
I have found that the calculation of the loss of generator is as following

    self.g_loss = -tf.reduce_sum(
            tf.reduce_sum(
                tf.one_hot(tf.to_int32(tf.reshape(self.x, [-1])), self.num_emb, 1.0, 0.0) * tf.log(
                    tf.clip_by_value(tf.reshape(self.g_predictions, [-1, self.num_emb]), 1e-20, 1.0)
                ), 1) * tf.reshape(self.rewards, [-1])
        )    

I am confused of the `g_predictions` here, which seems to be the softmax output of the generator. And `x` is the samples generated based on the `g_predictions`. So why should they be summed together? Any help is appreciated! Thanks in advance!
